### ISSUE TITLE
No WALs can be directly redone in page server for pg_bench simple_update?

### ISSUE THREAD

**Issue by skyzh** [neondatabase/neon#4349]
## Steps to reproduce

run pg_bench simple update

## Expected result

Some WALs can be redone within page server so as to improve performance

## Actual result

<img width="749" alt="image" src="https://github.com/neondatabase/neon/assets/4198311/ed340b71-8d37-4051-bbf8-9bc1d89c29aa">

this is computed using `pageserver_wal_redo_seconds` (which is updated for all redos) and `pageserver_wal_redo_records_histogram` (which is updated only for Postgres redos). It seems that no batches are applied within page server. This might not be a bug but I feel it is strange that we cannot redo anything in page server under this workload as it requires IPC.

## Logs, links
-

**Comment by hlinnaka** [comment]
@skyzh is this issue still relevant? If not, please close. If it is, can you try to ELI5 what the problem is, I didn't understand it?
